File: 프로그래머스/lv0/120924. 다음에 올 숫자/다음에 올 숫자.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> common)
{
    int answer = 0;
    int A = 0;
    for (int i = common.size()-1; i > 0; i--)
        A += common[i] - common[i-1];

    if (A / (common.size() - 1) == common[common.size()-1] - common[common.size()-2])
        answer = common[common.size()-1] + (A / (common.size() - 1));
    else
        answer = common[common.size()-1] * common[common.size() - 1] / common[common.size()-2];
    
    return answer;
}

File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.java
class Solution {

    public int solution(int[] common) {
        int diff = common[1] - common[0];
        int last = common[common.length - 1];

        if (common[2] - common[1] == diff) {
            return last + diff;
        }

        int diffs = common[2] / common[1];
        int lasts = last * diffs;
        if (last < 0) {
            return lasts * -1;
        } else {
            return lasts;
        }
    }
}